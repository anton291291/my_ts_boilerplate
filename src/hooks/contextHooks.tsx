import React, { useState, createContext } from 'react';
export const GridContext = createContext({} as GridContextType);

export declare namespace GlobalState {
    type CellItemType = { index: number; isClicked: boolean };

    type AxisType = {
        axis: {
            x: number;
            y: number;
        };
    };

    type CellType = {
        cells: Array<CellItemType>;
    };

    type IntervalIDType = {
        intervalID: number;
    };

    type isPlayType = {
        isPlay: boolean;
        speed: number;
    };

    type FormType = {
        name: string;
    };

    type GenType = {
        gen: number;
    };
}

export type StateType = GlobalState.AxisType &
    GlobalState.CellType &
    GlobalState.IntervalIDType &
    GlobalState.GenType &
    GlobalState.isPlayType &
    GlobalState.FormType;

type GridContextType = {
    state: StateType;
    setState: React.Dispatch<React.SetStateAction<StateType>>;
};

export const GridProvider = ({ children }) => {
    const y = 15;
    const x = 30;

    const [state, setState] = useState({
        axis: { x: x, y: y },
        cells: Array.from({ length: y * x }, (item, index) => ({
            index: index,
            isClicked: false
        })),
        gen: 1,
        name: '',
        intervalID: null,
        isPlay: false,
        speed: 1
    });

    return (
        <GridContext.Provider value={{ state, setState }}>
            {children}
        </GridContext.Provider>
    );
};
