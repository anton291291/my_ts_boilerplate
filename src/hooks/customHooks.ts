const useLogin = (params) => {
const [state, setstate] = useState(initialState)
}