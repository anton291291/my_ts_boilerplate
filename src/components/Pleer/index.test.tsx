import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { GridProvider } from '../../hooks/contextHooks';

describe('Pleer', () => {
    const output = mount(
        <GridProvider>
            <div />
        </GridProvider>
    );

    test('should render correctly', () => {
        expect(toJson(output)).toMatchSnapshot();
    });

    test('', () => {
        console.log(toJson(output));
    });
});
