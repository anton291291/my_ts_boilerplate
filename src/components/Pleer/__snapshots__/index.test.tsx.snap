// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pleer should render correctly 1`] = `
<GridProvider>
  <div />
</GridProvider>
`;
